File: backend/app/schemas/alert.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.utils.enums import AlertType, AlertSeverity

class AlertResponse(BaseModel):
    id: str
    patientId: Optional[str]
    type: AlertType
    severity: AlertSeverity
    message: str
    metadata: Optional[dict]
    acknowledged: bool
    createdAt: datetime
    resolvedAt: Optional[datetime]

    model_config = {"from_attributes": True}

class AlertListResponse(BaseModel):
    alerts: List[AlertResponse]
    totalCount: int
    unacknowledgedCount: int

File: frontend/types/alert.ts
export type AlertSeverity = 'CRITICAL' | 'WARNING' | 'INFO';
export type AlertType = 'DETERIORATION' | 'WAITING_BREACH' | 'CAPACITY' | 'SYSTEM';

export interface Alert {
  id: string;
  patientId: string | null;
  type: AlertType;
  severity: AlertSeverity;
  message: string;
  metadata: Record<string, unknown> | null;
  acknowledged: boolean;
  createdAt: string;
  resolvedAt: string | null;
}

export interface AlertListResponse {
  alerts: Alert[];
  totalCount: number;
  unacknowledgedCount: number;
}
